pkg/management/postgres/initdb.go

```diff
@@ -341,6 +324,33 @@ func (info InitInfo) ConfigureNewInstance(instance *Instance) error {
 	if err = info.executeSQLRefs(dbTemplate, info.PostInitTemplateSQLRefsFolder); err != nil {
 		return fmt.Errorf("could not execute post init application SQL refs: %w", err)
 	}
+
+	filePath := filepath.Join(info.PgData, constants.CheckEmptyWalArchiveFile)
+	// We create the check empty wal archive file to tell that we should check if the
+	// destination path it is empty
+	if err := fileutils.CreateEmptyFile(filepath.Clean(filePath)); err != nil {
+		return fmt.Errorf("could not create %v file: %w", filePath, err)
+	}
+
+	if info.ApplicationUser == "" {
+		return nil
+	}
```

Comment:
Does it make sense that if the ApplicationDatabase is empty we create the ApplicationRole but not viceversa?
Logic would be easier to require both/neither imo.